### PR TITLE
Expand SIFT test coverage

### PR DESCRIPTION
**Pull request by ahojnnes**
## Summary
- Expand test coverage for `feature/sift.cc` (93.1% coverage, 47 uncovered lines)
- Add tests for previously uncovered SIFT edge cases
- Targets item #15 from test coverage opportunities

## Test plan
- [ ] CI passes all existing and new tests
- [ ] New tests build and run successfully

### CODE REVIEW

File: src/colmap/feature/sift_test.cc [hunk 1]
```diff
@@ -1214,5 +1217,651 @@ TEST(MatchGuidedSiftFeaturesGPU, EssentialMatrix) {
...
+}
+
+TEST(LoadSiftFeaturesFromTextFile, NonexistentFile) {
+  FeatureKeypoints keypoints;
+  FeatureDescriptors descriptors;
+  EXPECT_THROW(
+      LoadSiftFeaturesFromTextFile("/nonexistent/path.txt", &keypoints,
+                                   &descriptors),
+      std::invalid_argument);
+}
```
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/colmap/feature/sift_test.cc:1760]: Keeping tests self-contained for clarity.

File: src/colmap/feature/sift_test.cc [hunk 2]
```diff
@@ -1214,5 +1217,651 @@ TEST(MatchGuidedSiftFeaturesGPU, EssentialMatrix) {
...
+  matcher_normal->Match(image1, image2, &matches_normal);
+  EXPECT_EQ(matches_normal.size(), 2);
+
+  // With a very small max_distance, no matches pass.
+  options.sift->max_distance = 0.001;
+  auto matcher_strict = CreateSiftFeatureMatcher(options);
+  FeatureMatches matches_strict;
+  matcher_strict->Match(image1, image2, &matches_strict);
+  EXPECT_EQ(matches_strict.size(), 0);
+}
```
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/colmap/feature/sift_test.cc:1891]: The random descriptors and reversed descriptors produce predictable distance properties (reversed descriptors have large L2 distance from originals). This is sufficient for testing the distance filter threshold.
